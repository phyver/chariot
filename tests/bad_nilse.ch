-- original example from Thorsten and Nilse

codata stream('x) where
    | Head : stream('x) -> 'x
    | Tail : stream('x) -> stream('x)

-- this type should be empty!
data tree where
    | Node : stream(tree) -> tree

val bad : stream(tree)
  | bad.Head = Node bad
  | bad.Tail = bad

val t : tree
  | t = Node bad

:show bad

