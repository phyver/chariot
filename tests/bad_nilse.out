>> option verbose is set to 1
>> coinductive type stream was succesfully defined
>> inductive type tree was succesfully defined
>> Typing for bad successful
>> the definition for bad is complete
!! warning: the definition bad is NOT provably total (weight_bound: 2, depth_bound: 2)

>> Typing for t successful
>> the definition for t is complete
>> the definition t is provably correct

val
    bad : stream(tree)
  | bad.Head¹ = Node² bad
  | bad.Tail¹ = bad

